File: src/main/java/ru/ezhov/jpa/editor/domain/model/OrderItem.java
package ru.ezhov.jpa.editor.domain.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.math.BigDecimal;

@Entity
@Table(name = "ORDER_ITEM")
public class OrderItem {
    @Id
    private int id;

    @ManyToOne()
    @JoinColumn(name = "ORDER_ID")
    private Order order;

    @OneToOne
    private Product product;

    @Column(name = "UNIT_PRICE")
    private BigDecimal unitPrice;

    @Column(name = "QUANTITY")
    private int quantity;

    @Override
    public String toString() {
        return "OrderItem{" +
                "id=" + id +
                ", orderId=" + order.getId() +
                ", productId=" + product.getId() +
                ", product=" + product +
                ", unitPrice=" + unitPrice +
                ", quantity=" + quantity +
                '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Order getOrder() {
        return order;
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public BigDecimal getUnitPrice() {
        return unitPrice;
    }

    public void setUnitPrice(BigDecimal unitPrice) {
        this.unitPrice = unitPrice;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: src/main/kotlin/ru/ezhov/jpa/editor/domain/model/OrderItem.kt
package ru.ezhov.jpa.editor.domain.model

import java.math.BigDecimal
import javax.persistence.*


@Entity
@Table(name = "ORDER_ITEM")
data class OrderItem(
        @Id
        var id: Int,

        @OneToOne
        var product: Product? = null,

        @Column(name = "UNIT_PRICE")
        var unitPrice: BigDecimal? = null,

        @Column(name = "QUANTITY")
        var quantity: Int = 0
) {
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ORDER_ID")
    lateinit var order: Order
}
